.github/workflows/precommit.yml

```diff
@@ -36,15 +36,19 @@ jobs:
       CXX: ${{ matrix.cxx }}
 
     steps:
+      - name: Symlink LLVM+libc++ (Linux LLVM+libc++ only)
+        if: startsWith(matrix.name, 'Linux LLVM+libc++')
+        run: sudo ln -s /usr/lib/llvm-10/include/c++/v1 /usr/include/c++/v1
+
       - uses: actions/checkout@v3
-      - name: install infrastructure (Linux)
-        if: matrix.os == 'ubuntu-20.04'
-        run: |
-          sudo apt-get install python3-dev python3-numpy python3-packaging
-          sudo ln -s /usr/lib/llvm-10/include/c++/v1 /usr/include/c++/v1
-      - name: install infrastructure (MacOS)
-        if: matrix.os == 'macos-10.15'
-        run: sudo -H pip3 install numpy packaging
+
+      - uses: actions/setup-python@v3
+        with:
+          python-version: '3.10'
```

Comment:
I'm not sure we have a use for Py 3.10, do we? We just use the runner's default Python version. If we wanted to test multiple versions, we should set up a matrix like in make_wheel.yml, but I'm not entirely sure that's warranted. We don't have a lot of Python code, and so I'm not sure we benefit from another multiplicative increase of test configurations.

Reply:
Changed to specify 3.x rather than 3.10.